include/faabric/scheduler/Scheduler.h

```diff
@@ -169,6 +172,11 @@ class Scheduler
 
     std::unordered_map<uint32_t, std::promise<int32_t>> threadResults;
 
+    std::unordered_map<std::string, faabric::scheduler::FunctionCallClient>
```

Comment:
A big design decision all around has been what to do with clients: persist them or open/close them.
* By persisting them we would expect a certain performance gain (assumption, not fact).
* By opening and closing them we have easier code to manage.
Message clients (aka 0MQ sockets) are _not_ thread safe and _must_ be opened and closed from the same thread.

I have followed the (always wrong) path of optimizing before measuring, having three different implementations:
* Thread-local, persistent, clients in `MpiWorld`. This handle _tons_ of messages, so it _theoretically_ made sense to try and optimize.
* Persistent clients in the scheduler.
* One-off clients in the state: only for the `inmemory` version.

I am open to re-considering any of this design choices.